tasty-query/shared/src/main/scala/tastyquery/reader/classfiles/ClassfileParser.scala

```diff
@@ -227,16 +228,36 @@ private[reader] object ClassfileParser {
       if cls.name == tpnme.Object then defn.createObjectMagicMethods(cls)
       else if cls.name == tpnme.String then defn.createStringMagicMethods(cls)
 
-    for (sym, sigOrDesc) <- loadMembers() do
-      sigOrDesc match
-        case SigOrDesc.Desc(desc) => sym.withDeclaredType(Descriptors.parseDescriptor(sym, desc))
-        case SigOrDesc.Sig(sig)   => sym.withDeclaredType(JavaSignatures.parseSignature(sym, sig, allRegisteredSymbols))
+    for (sym, javaFlags, sigOrDesc) <- loadMembers() do
+      val parsedType = sigOrDesc match
+        case SigOrDesc.Desc(desc) => Descriptors.parseDescriptor(sym, desc)
+        case SigOrDesc.Sig(sig)   => JavaSignatures.parseSignature(sym, sig, allRegisteredSymbols)
+      val adaptedType =
+        if sym.is(Method) && javaFlags.isVarargsIfMethod then patchForVarargs(sym, parsedType)
+        else parsedType
+      sym.withDeclaredType(adaptedType)
 
     for sym <- allRegisteredSymbols do sym.checkCompleted()
 
     innerClasses.declarations
   }
 
+  private def patchForVarargs(sym: TermSymbol, tpe: Type)(using Context): Type =
+    tpe match
+      case tpe: MethodType if tpe.paramNames.sizeIs >= 1 =>
+        defn.internalRepeatedAnnotClass match
+          case Some(annotClass) =>
+            val patchedLast = AnnotatedType(tpe.paramTypes.last, TQAnnotation(annotClass))
+            tpe.derivedLambdaType(tpe.paramNames, tpe.paramTypes.init :+ patchedLast, tpe.resultType)
+          case None =>
+            // Warn here? How?
+            tpe
```

Comment:
This seems like it should be an error, its incorrectly typed otherwise

Reply:
But if we do that, we cannot load Java/Scala 2-only classpaths anymore, which would be a pity.

Reply:
Well ... even if that's what it does, it doesn't match what it does at definition site for Scala 2 and 3. I think we should be more consistent.

Reply:
ok